metadata-ingestion/src/datahub/ingestion/transformer/extract_dataset_tags.py

```diff
@@ -0,0 +1,61 @@
+import re
+from enum import Enum
+from typing import List, Optional, cast
+
+from datahub.configuration.common import TransformerSemanticsConfigModel
+from datahub.emitter.mce_builder import Aspect
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.transformer.dataset_transformer import DatasetTagsTransformer
+from datahub.metadata.schema_classes import GlobalTagsClass, TagAssociationClass
+from datahub.utilities.urns.dataset_urn import DatasetUrn
+
+
+class ExtractTagsOption(Enum):
+    URN = "urn"
+
+
+class ExtractDatasetTagsConfig(TransformerSemanticsConfigModel):
+    extract_tags_from: ExtractTagsOption = ExtractTagsOption.URN
+    extract_tags_regex: str = ".*"
```

Comment:
probably doesn't make sense to have a default here - this should be a required config